src/chem/chem_system.cc

```diff
@@ -511,3 +531,554 @@ double ChemSystem::get_s2() const {
   }
   return s2;
 }
+
+//=====================================================
+MatrixXd ChemSystem::get_1rdm() const {
```

Comment:
Seems really long, better to put into a separate module.

Reply:
Done